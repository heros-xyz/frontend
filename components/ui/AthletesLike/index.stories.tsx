import type { Meta, StoryObj } from "@storybook/react";
import AthletesLike from ".";

const meta: Meta<typeof AthletesLike> = {
  title: "Components/AthletesLike",
  component: AthletesLike,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};
export default meta;
type Story = StoryObj<typeof AthletesLike>;

export const AthletesLikeStory: Story = {
  args: {
    data: [
      {
        id: "1",
        avatar:
          "https://images.pexels.com/photos/127160/pexels-photo-127160.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500",
        fullName: "Athlete Name 1",
        createdAt: "",
        sportName: "",
      },
    ],
    title: "Athletes you might also like:",
  },
};
