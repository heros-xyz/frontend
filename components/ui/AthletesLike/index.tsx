import { AspectRatio, Box, Grid, Text } from "@chakra-ui/react";
import { IAthleteInfo } from "@/types/athlete/types";
import HerosImage from "@/components/common/HerosImage";

interface AthletesLikeProp {
  data: IAthleteInfo[];
  title: string;
  onClick: (id: string) => void;
}

const AthletesLike: React.FC<AthletesLikeProp> = ({ data, title, onClick }) => {
  return (
    <Box
      px={{ base: "5", xl: "30px" }}
      py={{ base: "7", xl: "30px" }}
      pt="7"
      bg="white"
    >
      <Text
        color="primary"
        fontWeight="bold"
        fontSize={{ xl: "xl" }}
        mb={{ base: "5", xl: "30px" }}
      >
        {title}
      </Text>

      <Grid
        templateColumns="repeat(2, 1fr)"
        w="full"
        columnGap={{ base: 3.5, xl: 5 }}
        rowGap={5}
      >
        {data.map((el) => (
          <Box
            rounded="12px"
            boxShadow="md"
            key={el?.id}
            position="relative"
            cursor="pointer"
            onClick={() => onClick(el?.id)}
          >
            <AspectRatio maxW="400px" ratio={210 / 265}>
              <HerosImage
                src={el?.avatar}
                widthSize={500}
                heightSize={500}
                width={"100%"}
                height={"100%"}
                borderRadius="10px"
              />
            </AspectRatio>
            <Box
              position="absolute"
              bottom="0"
              w="100%"
              style={{
                background:
                  "linear-gradient(to bottom, rgba(0, 0, 0, 0) 60.94%, rgba(0, 0, 0, 0.8) 80.73%, #000000 100%)",
              }}
              borderBottomRightRadius="xl"
              borderBottomLeftRadius="xl"
            >
              <Box w="fit-content" ml={2.5} mb={2.5}>
                <Text
                  fontSize={{ base: "14px", lg: "18px" }}
                  color="grey.100"
                  fontWeight="bold"
                  lineHeight={{ base: "20px", lg: "25px" }}
                  textTransform="capitalize"
                >
                  {el?.nickName}
                </Text>
                <Text
                  color={el?.sportName ? "grey.100" : "transparent"}
                  fontSize={{ base: "10px", lg: "16px" }}
                  fontWeight="500"
                  lineHeight="14px"
                  py={{ xl: 1.5 }}
                >
                  {el?.sportName || "_"}
                </Text>
              </Box>
            </Box>
          </Box>
        ))}
      </Grid>
    </Box>
  );
};

export default AthletesLike;
