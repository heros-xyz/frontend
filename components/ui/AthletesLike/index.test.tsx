import { render } from "@testing-library/react";
import AthletesLike from ".";

it("renders homepage unchanged", () => {
  const { container } = render(
    <AthletesLike
      title="hehe"
      data={[
        {
          id: "1",
          avatar:
            "https://images.pexels.com/photos/127160/pexels-photo-127160.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500",
          fullName: "Athlete Name 1",
          createdAt: "",
          sport: "",
        },
      ]}
      onClick={() => console.log("test")}
    />
  );
  expect(container).toMatchSnapshot();
});
