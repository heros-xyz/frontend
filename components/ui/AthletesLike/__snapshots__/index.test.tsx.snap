// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="css-go1kwr"
  >
    <p
      class="chakra-text css-l1w3zv"
    >
      hehe
    </p>
    <div
      class="css-1j9fi3e"
    >
      <div
        class="css-dfpqc0"
      >
        <img
          alt=""
          class="chakra-image__placeholder css-1nqnlj4"
          src="https://via.placeholder.com/210x265"
        />
        <div
          class="css-rzhx1g"
        >
          <div
            class="css-1p6uzmq"
          >
            <p
              class="chakra-text css-1yzyoi4"
            />
            <p
              class="chakra-text css-4uwlir"
            >
              _
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
